tests/test_app.py

```diff
@@ -1003,3 +1005,355 @@ async def test_importer_mappings():
             assert resp.status == 400
             text = await resp.text()
             assert f"must provide file_list field. Your provided qs: {unquote(qsd)}" in text
+
+
+async def test_bulk_specification_success():
```

Comment:
These tests are long, but they're conceptually simple - just write some files, call the endpoint, and check the result. I tried moving some of the writes into separate functions, but to me it just made the tests harder to read since you have to jump all over the place in the code to figure out what's going on - it was simpler, at least to me, to just have a linear narrative for each test. Plus the contents change often and that's what takes up most of the lines.

Reply:
LGTM